src/cdomains/valueDomain.ml

```diff
@@ -562,11 +562,12 @@ struct
     | (x, `Bot) -> x
     | (`Int x, `Int y) -> (try `Int (ID.widen x y) with IntDomain.IncompatibleIKinds m -> Messages.warn ~category:Analyzer "%s" m; `Top)
     | (`Float x, `Float y) -> `Float (FD.widen x y)
+    (* TODO: symmetric widen, wtf? *)
     | (`Int x, `Address y)
     | (`Address y, `Int x) -> `Address (match ID.to_int x with
```

Comment:
I think this actually occurs quite often, because we sometimes try very hard to maintain addresses, and sometimes the path meaning we treat a certain itn like an address may only become live in later iterations.

Reply:
These do occur, e.g. when the old value is `Int`, the new value is `Address` and their `join` gives `Address`, then `widen` is from the old `Int` to the joined `Address`.
The change at https://github.com/goblint/analyzer/pull/809#discussion_r936328669 also confirms that they happen.